tools/Config.mk

```diff
@@ -531,3 +531,22 @@ else
   ARCHXXINCLUDES += ${shell $(INCDIR) -s "$(CC)" $(TOPDIR)$(DELIM)include$(DELIM)cxx}
 endif
 ARCHXXINCLUDES += ${shell $(INCDIR) -s "$(CC)" $(TOPDIR)$(DELIM)include}
+
+# If CC_TYPE is not set, try to guess.
+ifeq ($(CC_TYPE),)
```

Comment:
it isn't good to place toolchain specific flag here, the better place is Toolchain.defs or Make.defs

Reply:
i agree.
i just got tired to add this to every Make.defs and is trying to cheat. :-)

Reply:
how do you think about having, say, gcc.mk somewhere, so that every gcc-using Make.defs or Toolchain.defs can include it?

Reply:
I plan to move all toolchain specific option to arch/**/Toolchain.defs, I think it's the best place what I can imagine. Once the movement is done, all board will automatically support many good feature(e.g. uClibc++/libc++) without the boring configuration.

Reply:
sounds like a plan

Reply:
No, I will do it in this weekend.

Reply:
i meant, it sounds like a good plan. thank you for taking care of this.